compiler/src/test/scala/org/scalajs/core/compiler/test/ScalaJSDefinedTest.scala

```diff
@@ -691,4 +691,39 @@ class ScalaJSDefinedTest extends DirectTest with TestHelpers {
     """
   }
 
+  @Test
+  def overrideJavaLangObjectMethod: Unit = {
+    val objs = List("class", "object")
+
+    for {
+      obj <- objs
+    } {
+      s"""
+      $obj A extends js.Object {
+        override def clone(): Object = null
+        override def finalize(): Unit = ()
```

Comment:
I wonder if we shouldn't allow `clone()` and `finalize()` nevertheless. Since they are `protected`, they can never be called on an `Object`. They can only be called on a subclass that redefines them to be public. So in practice the "dangerous" behavior of an `override` not actually overriding some behavior cannot be observed. Therefore, it doesn't seem so dangerous.

The only real danger is overriding `hashCode()` and `equals(Any)`, because there the false overriding relationship can actually be observed. Perhaps we should target the warning at those two methods, and specifically mention them in the warning message?